main.go

```diff
@@ -151,7 +151,11 @@ func visit(url *url.URL) {
 	if (httpMethod == "POST" || httpMethod == "PUT") && postBody == "" {
 		log.Fatal("must supply post body using -d when POST or PUT is used")
 	}
-	req, err := http.NewRequest(httpMethod, url.String(), strings.NewReader(postBody))
+	body, err := createBody(postBody)
```

Comment:
If you handle the error inside `createBody` with a `log.Fatal{,f}` then the caller gets much simpler
```go
req, err := http.NewRequest(httpMethod, url.String(), createBody(postBody))
```
And there are less variables called `body`_something_ in the local scope.

I think this is worth breaking the separation of concerns by allowing `createBody` to call `log.Fatal` and exit the program early. There will certainly be other exceptions like this.